storage/store.go

```diff
@@ -1377,26 +1376,31 @@ func (s *Store) resolveWriteIntentError(ctx context.Context, wiErr *proto.WriteI
 	// We pushed the transaction(s) successfully, so resolve the intent(s).
 	for i, intent := range wiErr.Intents {
 		pushReply := bReply.Responses[i].InternalPushTxn
+		intentKey := intent.Key
 		resolveArgs := &proto.InternalResolveIntentRequest{
 			RequestHeader: proto.RequestHeader{
 				// Use the pushee's timestamp, which might be lower than the
 				// pusher's request timestamp. No need to push the intent higher
 				// than the pushee's txn!
 				Timestamp: pushReply.PusheeTxn.Timestamp,
-				Key:       intent.Key,
+				Key:       intentKey,
 				User:      UserRoot,
 				Txn:       pushReply.PusheeTxn,
 			},
 		}
 		resolveReply := &proto.InternalResolveIntentResponse{}
-		// Add resolve command with wait=false to add to Raft but not wait for completion.
-		waitForResolve := wait && i == len(wiErr.Intents)-1
-		if resolveErr := rng.AddCmd(ctx, proto.Call{Args: resolveArgs, Reply: resolveReply}, waitForResolve); resolveErr != nil {
-			if log.V(1) {
-				log.Warningc(ctx, "resolve for key %s failed: %s", intent.Key, resolveErr)
+
+		go func() {
+			if s.stopper.StartTask() {
+				resolveErr := rng.AddCmd(ctx, proto.Call{Args: resolveArgs, Reply: resolveReply})
```

Comment:
Yeah, that's definitely unfortunate. I just pushed another commit that plumbs a WaitGroup into `addWriteCmd` for this purpose. PTAL